src/renderer/src/scenes/main-scene/MainScene.tsx

```diff
@@ -173,20 +173,26 @@ const QueueModal: Component = () => {
   let queueModal: HTMLDivElement | undefined;
 
   const handleOutsideClick = (event: MouseEvent) => {
-    if (queueModal && !queueModal.contains(event.target as Node)) {
-      toggleSongQueueModalOpen();
+    if (event.target instanceof HTMLElement) {
+      if (
+        queueModal &&
+        !queueModal.contains(event.target as Node) &&
+        event.target.closest(".popover-overlay") === null
```

Comment:
Not too sure about this but I guess it'll do for now. Just have to rework all of this logic at some point I think

Reply:
This was to solve a bug where clicking on the context menu (opened on a song in the queue) would close both the context menu and the queue without doing the clicked action. The main scene is being completely remade in #88 so this is a temporary fix.